pkg/controllers/kubernetes/extension.go

```diff
@@ -42,40 +50,37 @@ func serviceCreated(obj interface{}) {
 	}
 }
 
-func serviceUpdated(oldObj, newObj interface{}) {
+func (e extensionHandler) serviceUpdated(oldObj, newObj interface{}) {
 	oldService, newService := oldObj.(*v1.Service), newObj.(*v1.Service)
 	// If resourceVersion differs between old and new, an actual update event was observed
 	versionUpdated := oldService.ResourceVersion != newService.ResourceVersion
-	oldExtensions := len(router.GetExtensions())
 	// Updated services will still be in the same namespace
-	if oldService.Namespace == installNamespace {
-		if value := oldService.Labels[router.ExtensionLabelKey]; versionUpdated && value == router.ExtensionLabelValue {
+	if oldService.Namespace == e.installNamespace {
+		updates := 0
+		if value := oldService.Labels[router.ExtensionLabelKey]; versionUpdated && value == router.ExtensionLabelValue && oldService.Spec.ClusterIP != "" {
 			logging.Log.Debugf("Extension Controller Update: Removing old extension '%s'", oldService.Name)
-			router.UnregisterExtension(oldService)
+			e.UnregisterExtension(oldService)
+			updates++
 		}
-		if value := newService.Labels[router.ExtensionLabelKey]; versionUpdated && value == router.ExtensionLabelValue {
+		if value := newService.Labels[router.ExtensionLabelKey]; versionUpdated && value == router.ExtensionLabelValue && newService.Spec.ClusterIP != "" {
 			logging.Log.Debugf("Extension Controller Update: Add new extension '%s'", newService.Name)
-			router.RegisterExtension(newService)
+			e.RegisterExtension(newService)
+			updates += 2
```

Comment:
Why is this two? Based on the switch later are we basically using this number as a status code of sorts (if so I'd rather we didn't, use an enum like object so it's more readable)

Reply:
I am using those numbers like their bitwise representations so 1 = 01, 2 = 10. If it was to increment one it would be ambiguous which case was triggered in the situation where only a service is either now an extension or no longer an extension (the extension label is removed).

Reply:
Updated to use strings